File: 프로그래머스/0/181830. 정사각형으로 만들기/정사각형으로 만들기.cpp
#include <string>
#include <vector>

using namespace std;

vector<vector<int>> solution(vector<vector<int>> arr) {
    int colsize = arr.size();
    int rowsize = 0;
    
    for(int i = 0; i < arr.size(); i++)
    {
        if(arr[i].size() > rowsize)
            rowsize = arr[i].size();
    }
    
    int size = colsize > rowsize ? colsize : rowsize;
    vector<vector<int>> answer(size, vector<int>(size));
    
    for(int i = 0; i < arr.size(); i++)
    {
        for(int j = 0; j < arr[i].size(); j++)
            answer[i][j] = arr[i][j];
    }
    
    return answer;
}

File: 프로그래머스/0/181830. 정사각형으로 만들기/정사각형으로 만들기.cs
using System;

public class Solution {
    public int[,] solution(int[,] arr) {
        int colSize = arr.GetLength(0);
        int rowSize = arr.GetLength(1);
        int size = colSize > rowSize ? colSize : rowSize;
        int[,] answer = new int[size,size];
        for(int i = 0; i < arr.GetLength(0); i++)
        {
            for(int j = 0; j < arr.GetLength(1); j++)
                answer[i,j] = arr[i,j];
        }
        return answer;
    }
}